### ISSUE TITLE
Node has 0 tokens in accountant balance even after giving traffic

### ISSUE THREAD

**Issue by chompomonim** [mysteriumnetwork/node#1878]
In node's logs there are error messages: `failed to get exchange message: could not reveal R: could not reveal R for accountant: expected hashlock: \"5546c9925583842aea90b32ef3a314b510696ae24bb5f62476fb975388592cc1\": hashlock missmatch: hashlock missmatch` .

Node has 0 tokens in his accountant channel even after disabling recheck for revealing R. 
There are quite a few such cases already found.

```
 [0m[36m >[0m Proposal pinged for node: 0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f service type: openvpn
[90m2020-03-10T15:53:50.678[0m [33mDBG[0m [1msession/pingpong/price_calculator.go:66 [0m[36m >[0m Calculated price 736963. Time component: 402452, data component: 334511 
[90m2020-03-10T15:53:50.718[0m [33mDBG[0m [1mcommunication/nats/sender.go:60         [0m[36m >[0m Message "eae97a3c-42bb-48f8-acf2-c7cf7a7faeea.session-invoice" sending: {"payload":{"invoice":{"AgreementID":1122580110159445382,"AgreementTotal":736963,"TransactorFee":0,"Hashlock":"3e243bd7c5f40eca2feff52753f65b40dc251424a0d9b1530d29f4b090296c86","Provider":"0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f"}},"signature":"apkieiRQvRAhopUwZ91XB3UwA0nmX1hLiovyIQ58nkMDUq4kUQ9HZaL6ENjS6LVzlt7IZhH3Ytjf+52ikHxooQE="}
[90m2020-03-10T15:53:51.047[0m [33mDBG[0m [1mcommunication/nats/receiver.go:54       [0m[36m >[0m Message "eae97a3c-42bb-48f8-acf2-c7cf7a7faeea.session-exchange" received: {"payload":{"exchangeMessage":{"Promise":{"ChannelID":"roW10++JHEW7CxG8Ng3Vqev4STs=","Amount":934950494,"Fee":0,"Hashlock":"PiQ718X0Dsov7/UnU/ZbQNwlFCSg2bFTDSn0sJApbIY=","R":null,"Signature":"UqJDT2mYBkBOVXwZ1FSv7/Z645IcpSmR5YdQonNx4DcUS3zaX3faRLshOkYsXU7EFZ3Z8jWgjcPXY/TVuT+nCRw="},"AgreementID":1122580110159445382,"AgreementTotal":736963,"Provider":"0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f","Signature":"69f91e26ca464609d09c27af4a1f343d6919433431c0c06a6b3d972d7d316c545891bf4ad0e52e8fcc6b6f284acbc196f27cca329adaa4146e0b9d28b42ef8851c"}},"signature":"pPdzZbjNMG7bPnIw6jgzanZeC5LbEc2HkpFwmv6DbNo2yEyPEptAVbjWtRMATYF9yXgoCAFOZ0tMOYK5m9AvEAA="}
[90m2020-03-10T15:53:51.080[0m [31mWRN[0m [1msession/pingpong/invoice_tracker.go:510 [0m[36m >[0m accountant error, will retry [31merror=[0m[31m"could not reveal R for accountant: expected hashlock: \"5546c9925583842aea90b32ef3a314b510696ae24bb5f62476fb975388592cc1\": hashlock missmatch: hashlock missmatch"[0m
[90m2020-03-10T15:53:56.283[0m [32mINF[0m [1mmarket/mysterium/mysterium_api.go:203   [0m[36m >[0m Proposal pinged for node: 0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f service type: wireguard
DEBUG: [userspace-wg]2020/03/10 15:54:13 peer(hsZ5…RgQU) - Retrying handshake because we stopped hearing back after 15 seconds
DEBUG: [userspace-wg]2020/03/10 15:54:13 peer(hsZ5…RgQU) - Sending handshake initiation
DEBUG: [userspace-wg]2020/03/10 15:54:13 peer(hsZ5…RgQU) - Received handshake response
DEBUG: [userspace-wg]2020/03/10 15:54:13 peer(hsZ5…RgQU) - Sending keepalive packet
[90m2020-03-10T15:54:32.920[0m [32mINF[0m [1mmarket/mysterium/mysterium_api.go:203   [0m[36m >[0m Proposal pinged for node: 0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f service type: noop
DEBUG: [userspace-wg]2020/03/10 15:54:33 peer(hsZ5…RgQU) - Sending keepalive packet
[90m2020-03-10T15:54:50.747[0m [33mDBG[0m [1msession/pingpong/price_calculator.go:66 [0m[36m >[0m Calculated price 787366. Time component: 452509, data component: 334857 
[90m2020-03-10T15:54:50.782[0m [33mDBG[0m [1mcommunication/nats/sender.go:60         [0m[36m >[0m Message "eae97a3c-42bb-48f8-acf2-c7cf7a7faeea.session-invoice" sending: {"payload":{"invoice":{"AgreementID":1122580110159445382,"AgreementTotal":787366,"TransactorFee":0,"Hashlock":"0e01ea474d2a1873fe56353c9aa0e43f50fc2c90ef2ebef7dbe542d91e3a2fab","Provider":"0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f"}},"signature":"FBvoJ3i2Ixx3vib+LX2QZY3vsBBeZjIT9jywrnl+aTtoLO8N28HVHZmszc9RoGPFd/kdrjX1xqrT0e/YXAYL1wE="}
[90m2020-03-10T15:54:51.137[0m [33mDBG[0m [1mcommunication/nats/receiver.go:54       [0m[36m >[0m Message "eae97a3c-42bb-48f8-acf2-c7cf7a7faeea.session-exchange" received: {"payload":{"exchangeMessage":{"Promise":{"ChannelID":"roW10++JHEW7CxG8Ng3Vqev4STs=","Amount":935000897,"Fee":0,"Hashlock":"DgHqR00qGHP+VjU8mqDkP1D8LJDvLr732+VC2R46L6s=","R":null,"Signature":"R4upPDhNrLTMwoipb/GubxeCKVucsrA/vQFGQY+bQxoa+a3EiDwkVLUTeHXB3K0NPWL3kOtviDLjJ01Eeh2Yehw="},"AgreementID":1122580110159445382,"AgreementTotal":787366,"Provider":"0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f","Signature":"52b979c09858f2cbc10b4a04c80b423f38968f3e02133d1496d40970cfcc0a401ead85f882f55dd4b73e7951fa64d553527a7fdbaafa7cb02a1dc17ada254e1d1b"}},"signature":"R90i4KZVnHer8lCf6D+KtiHKTWmNw4+Vp042UHNzxol6WSkSdf+SNvZv4PPeac3sirhJpbtsqTiN4F/yEhjdLgA="}
[90m2020-03-10T15:54:51.173[0m [31mWRN[0m [1msession/pingpong/invoice_tracker.go:510 [0m[36m >[0m accountant error, will retry [31merror=[0m[31m"could not reveal R for accountant: expected hashlock: \"5546c9925583842aea90b32ef3a314b510696ae24bb5f62476fb975388592cc1\": hashlock missmatch: hashlock missmatch"[0m
DEBUG: [userspace-wg]2020/03/10 15:55:01 peer(hsZ5…RgQU) - Sending keepalive packet
DEBUG: [userspace-wg]2020/03/10 15:55:11 peer(hsZ5…RgQU) - Sending keepalive packet
DEBUG: [userspace-wg]2020/03/10 15:55:22 peer(hsZ5…RgQU) - Sending keepalive packet
DEBUG: [userspace-wg]2020/03/10 15:55:32 peer(hsZ5…RgQU) - Sending keepalive packet
[90m2020-03-10T15:55:50.824[0m [33mDBG[0m [1msession/pingpong/price_calculator.go:66 [0m[36m >[0m Calculated price 837461. Time component: 502573, data component: 334888 
[90m2020-03-10T15:55:50.859[0m [33mDBG[0m [1mcommunication/nats/sender.go:60         [0m[36m >[0m Message "eae97a3c-42bb-48f8-acf2-c7cf7a7faeea.session-invoice" sending: {"payload":{"invoice":{"AgreementID":1122580110159445382,"AgreementTotal":837461,"TransactorFee":0,"Hashlock":"a5efc1c087897867f2c30879b1612f50c4baafb2e97055e16dab619bd10af65e","Provider":"0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f"}},"signature":"dMwSgEjbXal/DDIXO1xRHN+Fq54TALTOMV+7lHRPahFoApjnGg3u+iM//1KPiMjgfvgEFnfEcOr1S6wKoARqwwA="}
[90m2020-03-10T15:55:51.454[0m [33mDBG[0m [1mcommunication/nats/receiver.go:54       [0m[36m >[0m Message "eae97a3c-42bb-48f8-acf2-c7cf7a7faeea.session-exchange" received: {"payload":{"exchangeMessage":{"Promise":{"ChannelID":"roW10++JHEW7CxG8Ng3Vqev4STs=","Amount":935050992,"Fee":0,"Hashlock":"pe/BwIeJeGfywwh5sWEvUMS6r7LpcFXhbathm9EK9l4=","R":null,"Signature":"By8w+1D3F1tvMBb4vyABgEDdusUWbejMMyD/ueABhM0GqtS+OexqDfbjCYu4+IhwtN6B0mCqYcJJCPhFniz4Vhs="},"AgreementID":1122580110159445382,"AgreementTotal":837461,"Provider":"0x04413fb0d8d9ddfdeb74245e3e49baa97b02f46f","Signature":"4ab9aa121dee3fc61566c903a619a2453492b0214ebb749a519120227c10d3373223efead2635ef43b6c7653ea641d53dc31a6d9315bc60f99cbce13850fe9a21b"}},"signature":"jdCYIoHyECJMQ0ImlfOFfPB/oEaN8DBNd5PZADtN5B5lovUfdyruVjnKRm/pNuIbEF8hE8ovst4bwyKn9X1DJwA="}
[90m2020-03-10T15:55:51.523[0m [1m[31mERR[0m[0m [1msession/manager.go:155                  [0m[36m >[0m Payment engine error [31merror=[0m[31m"failed to get exchange message: could not reveal R: could not reveal R for accountant: expected hashlock: \"5546c9925583842aea90b32ef3a314b510696ae24bb5f62476fb975388592cc1\": hashlock missmatch: hashlock missmatch"[0m
[90m2020-03-10T15:55:51.523[0m [33mDBG[0m [1msession/pingpong/invoice_tracker.go:591 [0m[36m >[0m Stopping...
[90m2020-03-10T15:55:51.523[0m [32mINF[0m [1mservices/wireguard/service/service_unix.go:215[0m[36m >[0m Cleaning up session 1f9e2261-917f-412d-b316-98e4a8f36e1a
[90m2020-03-10T15:55:51.523[0m [32mINF[0m [1mnat/service_iptables.go:75              [0m[36m >[0m Deleting NAT/Firewall rules
[90m2020-03-10T15:55:51.523[0m [32mINF[0m [1mservices/wireguard/service/stats_publisher.go:62[0m[36m >[0m Stopped publishing statistics for session 1f9e2261-917f-412d-b316-98e4a8f36e1a
[90m2020-03-10T15:55:51.527[0m [33mDBG[0m [1mnat/service_iptables.go:169             [0m[36m >[0m "/sbin/iptables -D PREROUTING --source 10.182.0.2/24 --destination 10.182.0.1 --protocol udp --dport 53 --jump REDIRECT --to-ports 11253 --table nat" output:

[90m2020-03-10T15:55:51.531[0m [33mDBG[0m [1mnat/service_iptables.go:169             [0m[36m >[0m "/sbin/iptables -D PREROUTING --source 10.182.0.2/24 --destination 10.182.0.1 --protocol tcp --dport 53 --jump REDIRECT --to-ports 11253 --table nat" output:

[90m2020-03-10T15:55:51.534[0m [33mDBG[0m [1mnat/service_iptables.go:169             [0m[36m >[0m "/sbin/iptables -D FORWARD --source 10.182.0.2/24 --destination 10.0.0.0/8 --jump DROP" output:

[90m2020-03-10T15:55:51.538[0m [33mDBG[0m [1mnat/service_iptables.go:169             [0m[36m >[0m "/sbin/iptables -D FORWARD --source 10.182.0.2/24 --destination 172.16.0.0/12 --jump DROP" output:

[90m2020-03-10T15:55:51.542[0m [33mDBG[0m [1mnat/service_iptables.go:169             [0m[36m >[0m "/sbin/iptables -D FORWARD --source 10.182.0.2/24 --destination 192.168.0.0/16 --jump DROP" output:

[90m2020-03-10T15:55:51.546[0m [33mDBG[0m [1mnat/service_iptables.go:169             [0m[36m >[0m "/sbin/iptables -D FORWARD --source 10.182.0.2/24 --destination 127.0.0.0/8 --jump DROP" output:

[90m2020-03-10T15:55:51.549[0m [33mDBG[0m [1mnat/service_iptables.go:169             [0m[36m >[0m "/sbin/iptables -D POSTROUTING --source 10.182.0.2/24 ! --destination 10.182.0.2/24 --jump SNAT --to 172.16.86.46 --table nat" output:

[90m2020-03-10T15:55:51.549[0m [32mINF[0m [1mnat/service_iptables.go:87              [0m[36m >[0m Deleting NAT/Firewall rules... done
INFO: [userspace-wg]2020/03/10 15:55:51 Device closing
DEBUG: [userspace-wg]2020/03/10 15:55:51 Routine: event worker - stopped
DEBUG: [userspace-wg]2020/03/10 15:55:51 Routine: TUN reader - stopped

```

**Comment by chompomonim** [comment]
Should be fixed after #1585 will be done.

**Comment by NeverVane** [comment]
It looks it might be solved
